File: trip-service/src/main/java/com/study/trip/application/dto/CreateTripRequest.java
package com.study.trip.application.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Getter
@NoArgsConstructor
public class CreateTripRequest {
    private Long userId;
    private String title;
    private String description;
    private LocalDate startDate;
    private LocalDate endDate;
}

File: trip-service/src/main/kotlin/com/study/trip/application/dto/CreateTripRequest.kt
package com.study.trip.application.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDate

data class CreateTripRequest(
    @field:NotBlank(message = "Title is required")
    val title: String = "",

    val description: String? = null,

    @field:NotNull(message = "Start date is required")
    val startDate: LocalDate? = null,

    @field:NotNull(message = "End date is required")
    val endDate: LocalDate? = null
)
